app/controllers/tasks_controller.rb

```diff
@@ -4,7 +4,31 @@ class TasksController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @tasks = current_user.tasks
-    render json: @user
+    # @task = current_user.tasks
+    @pagy, @tasks = pagy(Task.includes(:creator, :assignee, :parent_task, :subtasks).all)
+    render json: @tasks
+  end
+
+  def create
+    @task = Task.new(task_params)
+    if @task.save
+      render json: @task, status: :created, include: ['subtasks']
+    else
+      render json: { errors: @task.errors.full_messages.join(', ') }, status: :unprocessable_entity
```

Comment:
this might become a pattern for most create and update methods. Extract it to an error handler for reusability.